diff -Nur ffi-yajl-0.2.0.old/lib/ffi_yajl/ext.rb ffi-yajl-0.2.0/lib/ffi_yajl/ext.rb
--- ffi-yajl-0.2.0.old/lib/ffi_yajl/ext.rb	2014-06-27 02:20:31.000000000 +0000
+++ ffi-yajl-0.2.0/lib/ffi_yajl/ext.rb	2014-06-27 02:44:19.000000000 +0000
@@ -3,30 +3,10 @@
 require 'ffi_yajl/encoder'
 require 'ffi_yajl/parser'
 require 'ffi'
-require 'libyajl2'
-begin
-  require 'fiddle'
-rescue LoadError
-end
-begin
-  require 'dl'
-rescue LoadError
-end
 
 module FFI_Yajl
-  # FIXME: DRY with ffi_yajl/ffi.rb
-  libname = ::FFI.map_library_name("yajl")
-  libpath = File.expand_path(File.join(Libyajl2.opt_path, libname))
-  libpath.gsub!(/dylib/, 'bundle')
-  libpath = ::FFI.map_library_name("yajl") unless File.exist?(libpath)
-  if defined?(Fiddle) && Fiddle.respond_to?(:dlopen)
-    ::Fiddle.dlopen(libpath)
-  elsif defined?(DL) && DL.respond_to?(:dlopen)
-    ::DL.dlopen(libpath)
-  else
-    extend ::FFI::Library
-    ffi_lib libpath
-  end
+  extend ::FFI::Library
+  ffi_lib 'yajl'
 
   class Parser
     require 'ffi_yajl/ext/parser'
diff -Nur ffi-yajl-0.2.0.old/lib/ffi_yajl/ffi.rb ffi-yajl-0.2.0/lib/ffi_yajl/ffi.rb
--- ffi-yajl-0.2.0.old/lib/ffi_yajl/ffi.rb	2014-06-27 02:20:31.000000000 +0000
+++ ffi-yajl-0.2.0/lib/ffi_yajl/ffi.rb	2014-06-27 02:43:27.000000000 +0000
@@ -1,21 +1,11 @@
 require 'rubygems'
 
-require 'libyajl2'
 require 'ffi'
 
 module FFI_Yajl
   extend ::FFI::Library
 
-  libname = ::FFI.map_library_name("yajl")
-  libpath = File.expand_path(File.join(Libyajl2.opt_path, libname))
-  libpath.gsub!(/dylib/, 'bundle')
-
-  if File.file?(libpath)
-    # use our vendored version of libyajl2 if we find it installed
-    ffi_lib libpath
-  else
-    ffi_lib 'yajl'
-  end
+  ffi_lib 'yajl'
 
   class YajlCallbacks < ::FFI::Struct
     layout :yajl_null, :pointer,
