--- ffi-yajl-0.2.0/ffi-yajl.gemspec.old	2014-06-27 02:24:15.000000000 +0000
+++ ffi-yajl-0.2.0/ffi-yajl.gemspec	2014-06-27 02:25:35.000000000 +0000
@@ -29,16 +29,14 @@
       s.add_development_dependency(%q<pry>, ["~> 0.9"])
       s.add_development_dependency(%q<rake-compiler>, ["~> 0.8.3"])
       s.add_development_dependency(%q<mime-types>, ["~> 1.16"])
-      s.add_runtime_dependency(%q<ffi>, ["~> 1.5"])
-      s.add_runtime_dependency(%q<libyajl2>, [">= 0"])
+      s.add_runtime_dependency(%q<ffi>)
     else
       s.add_dependency(%q<rake>, ["~> 10.1"])
       s.add_dependency(%q<rspec>, ["~> 2.99"])
       s.add_dependency(%q<pry>, ["~> 0.9"])
       s.add_dependency(%q<rake-compiler>, ["~> 0.8.3"])
       s.add_dependency(%q<mime-types>, ["~> 1.16"])
-      s.add_dependency(%q<ffi>, ["~> 1.5"])
-      s.add_dependency(%q<libyajl2>, [">= 0"])
+      s.add_dependency(%q<ffi>)
     end
   else
     s.add_dependency(%q<rake>, ["~> 10.1"])
@@ -46,8 +44,7 @@
     s.add_dependency(%q<pry>, ["~> 0.9"])
     s.add_dependency(%q<rake-compiler>, ["~> 0.8.3"])
     s.add_dependency(%q<mime-types>, ["~> 1.16"])
-    s.add_dependency(%q<ffi>, ["~> 1.5"])
-    s.add_dependency(%q<libyajl2>, [">= 0"])
+    s.add_dependency(%q<ffi>)
   end
 end
 
